mqtt/mqtt-edgehub/src/command/handler.rs

```diff
@@ -70,14 +73,17 @@ impl CommandHandler {
             Duration::from_secs(60),
         );
 
-        let subscribe_topics = commands.keys().cloned().collect::<Vec<_>>();
-        subscribe(&mut client, subscribe_topics).await?;
-
-        Ok(CommandHandler {
-            broker_handle,
+        CommandHandler {
             client,
-            commands,
-        })
+            commands: HashMap::new(),
+        }
+    }
+
+    // TODO refactor and move it inside the [`run`] method
+    pub async fn init(&mut self) -> Result<(), CommandHandlerError> {
```

Comment:
Why not name this new()?

Reply:
`new` - create an instance of the handler, but then it takes commands one by one. only after all commands added we should ask mqtt client to subscribe to all topics.